### PR TITLE
Processing send txs 

### PR DESCRIPTION
**Pull request by mateodaza**

### CODE REVIEW

File: src/components/SendModal/SendConfirmation.tsx [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+import { useEffect, useState } from 'react'
+import { TokenItem } from '../TokenItem'
+import LoadingArc from './LoadingArc'
+import Image from 'next/image'
+import { truncateEthAddress } from '@/utils/strings'
+
+interface ISendConfirmation {
+  destination: string | null
+  amount: string | null
+  selectedToken: any
```
Comment (mmahdigh) [inline src/components/SendModal/SendConfirmation.tsx:14]: I think we can use the `ICryptoToken` type here. However, we need to add a "symbol" property to it too.

File: src/components/SendModal/SendConfirmation.tsx [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+  destination,
+  selectedToken,
+  amount,
+}: ISendConfirmation) => {
+  const router = useRouter()
+  const [txLoading, setTxLoading] = useState<boolean>(false)
+  const [txComplete, setTxComplete] = useState(false)
+  const [txFailed, setTxFailed] = useState(false)
+  const [gasCost, setGasCost] = useState<string | null>('')
+  const [successMessage, setSuccessMessage] = useState<string | null>(null)
```
Comment (mmahdigh) [inline src/components/SendModal/SendConfirmation.tsx:33]: `successMessage` is not used anywhere?

File: src/components/SendModal/SendConfirmation.tsx [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+const SendConfirmation = ({
+  onDismiss,
+  setConfirmTx,
+  setTxDone,
+  destination,
+  selectedToken,
+  amount,
+}: ISendConfirmation) => {
+  const router = useRouter()
+  const [txLoading, setTxLoading] = useState<boolean>(false)
```
Comment (mmahdigh) [inline src/components/SendModal/SendConfirmation.tsx:29]: I'm unsure if what I'm saying will apply, but can we use an enum here? Like:


```ts
enum TxState {
  Initial,
  Loading,
  Complete,
  Failed,
}

const [txState, setTxState] = useState<TxState>(TxState.Initial)
```

File: src/components/SendModal/SendConfirmation.tsx [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+
+  if (txFailed) {
+    return (
+      <div className="flex flex-wrap gap-4 flex-col align-center justify-center text-center gap-x-16">
+        <Typography fontVariant="headingThree" weight="bold">
+          Transaction Failed
+        </Typography>
+
+        <Button
+          onClick={async () => {
```
Comment (mmahdigh) [inline src/components/SendModal/SendConfirmation.tsx:84]: We can probably do `onClick={sendTx}` here.

File: src/components/SendModal/index.tsx [hunk 5]
```diff
@@ -72,135 +101,132 @@ export const SendModal: React.FC<{
...
+                      width="auto"
+                      onClick={() => setSelectedToken(null)}>
+                      <ChevronRight />
+                    </IconButton>
+                  </div>
+                  <Input
+                    label="Amount"
+                    placeholder="0.0"
+                    clearable
+                    onChange={(e: any) => {
```
Comment (mmahdigh) [inline src/components/SendModal/index.tsx:172]: I don't think the `any` is needed here?

**Review by mmahdigh — APPROVED** [review]

**Review by jainkrati — APPROVED** [review]

**Pull request merged**